spec/services/create_mail_method_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+describe CreateMailMethod do
+  describe '#call' do
+    let(:mail_method) { Spree::MailMethod.create(environment: 'test') }
+    let(:mail_settings) { instance_double(Spree::Core::MailSettings) }
+    let(:attributes) do
+      { preferred_smtp_username: "smtp_username", environment: "test" }
+    end
+
+    before do
+      allow(Spree::MailMethod)
+        .to receive(:create).with(environment: 'test').and_return(mail_method)
+      allow(Spree::Core::MailSettings).to receive(:init) { mail_settings }
+    end
+
+    it 'creates a new MailMethod' do
+      described_class.new(attributes).call
+
+      expect(Spree::MailMethod)
+        .to have_received(:create).with(environment: 'test') { mail_method }
+    end
+
+    it 'updates the MailMethod' do
+      expect(mail_method)
+        .to(receive(:update_attributes)).with(attributes) { mail_method }
+
+      described_class.new(attributes).call
```

Comment:
Why this test structure is different from the others?

Reply:
that's because this one is actually hitting the DB. I'll try to make it look like the others.